scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1333,24 +1333,49 @@ static void task_uncharge_qrt(struct task_ctx *taskc)
 static void kick_idle_cpu(struct task_struct *p, struct layer *layer)
 {
 	const struct cpumask *idle_smtmask;
+	struct bpf_cpumask *cand_cpumask;
+	struct task_ctx *taskc;
+	struct cpu_ctx *cpuc;
 	s32 cpu;
 
 	if (!(idle_smtmask = scx_bpf_get_idle_smtmask()))
 		return;
-
+	
+	if (!(cpuc = lookup_cpu_ctx(-1)) || !(cand_cpumask = bpf_cpumask_create())) {
+		scx_bpf_put_idle_cpumask(idle_smtmask);
+		return;
+	}
+	
 	if (layer->kind == LAYER_KIND_CONFINED) {
 		const struct cpumask *layer_cpumask;
 
-		if (!(layer_cpumask = lookup_layer_cpumask(layer->id)))
-			goto out;
-		cpu = pick_idle_cpu_from(layer_cpumask, 0, idle_smtmask, layer);
+		if (!(layer_cpumask = lookup_layer_cpumask(layer->id))) {
+			bpf_cpumask_release(cand_cpumask);
+			scx_bpf_put_idle_cpumask(idle_smtmask);
+			return;
+		}
+
+		if (layer->skip_remote_node && (taskc = lookup_task_ctx(p)) && taskc->layered_node_mask) {
```

Comment:
`kick_idle_cpu` is only called from `enqueue` where `taskc` is already available. Why not pass it as a arg instead of doing a double map lookup every enqueue?